connection_producer.go

```diff
@@ -149,8 +150,14 @@ func (c *snowflakeConnectionProducer) Connection(ctx context.Context) (interface
 		return nil, connutil.ErrNotInitialized
 	}
 
+	// If we already have a DB, test it and return
 	if c.snowflakeDB != nil {
-		return c.snowflakeDB, nil
+		if err := c.snowflakeDB.PingContext(ctx); err == nil {
+			return c.snowflakeDB, nil
+		}
+		// If the ping was unsuccessful, close it and ignore errors as we'll be
+		// reestablishing anyways
+		c.snowflakeDB.Close()
```

Comment:
I want to make sure I understand: is this line what effectively fixes the issue? We received an error from the ping because the authentication expired, so we are explicitly closing the connection, which will then be re-established on the next request.